libs/platform/CMakeLists.txt

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: MIT
+
+#
+# ebpf_program_types_interface
+#
+
+find_program(midl_compiler_path "midl" REQUIRED)
+
+set(ebpf_program_types_h_name "ebpf_program_types_h.h")
+set(ebpf_program_types_h_path "${CMAKE_CURRENT_BINARY_DIR}/${ebpf_program_types_h_name}")
+
+set(ebpf_program_types_c_name "ebpf_program_types_c.c")
+set(ebpf_program_types_c_path "${CMAKE_CURRENT_BINARY_DIR}/${ebpf_program_types_c_name}")
+
+# TODO(Alessandro): It's not ideal to generate files in the source directory, but I have
```

Comment:
if you propose to check in a file with a TODO in it, then file a github issue and reference the issue instead of your name in the TODO comment.  Or, just do the todo :)

Reply:
Removing the in-source build parts would require changes in the Visual Studio solution, which I'm trying not to alter in this PR. I think I can remove the TODO for now since it's bahaving like VS for now